main.c

```diff
@@ -399,10 +422,11 @@ STATIC bool run_code_py(safe_mode_t safe_mode, bool *simulate_reset) {
         };
         #endif
 
+        #if MICROPY_ENABLE_PYSTACK
+        supervisor_allocation *pystack = alloc_pystack();
+        #endif
         supervisor_allocation *heap = allocate_remaining_memory();
-
-        // Prepare the VM state.
-        start_mp(heap);
+        start_mp(heap, pystack);
```

Comment:
The problem here is that `pystack` is not defined if not MICROPY_ENABLE_PYSTACK, which is a compile error.

Unfortunately the #define really makes a mess of what is otherwise simple code. Idk, the best solution, but you might consider creating a struct that contains heap and pystack if enabled. Return this struct from `start_mp` and pass it to `stop_mp`. This way all the MICROPY_ENABLE_PYSTACK nonsense is contained to this struct and those functions.

Reply:
An excellent idea, doing it.

Reply:
Implemented struct stacks that contains pystack and heap.
If stackless only heap.

Reply:
My preference would to be always passing in a pystack and having it NULL when disabled.

Reply:
```
#else
pystack = NULL;
```